Make fast64 even faster

bkakadiya42 opened the pull request.

Commit:
Use StandardJS, cleanup tests.

Commit:
Add a benchmark test to compare it with the native implementation.

ref. https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding\#The_Unicode_Problem

Commit:
Hardcode the lookup/rlookup table to speed up the first encode/decode calls.

Commit:
Make the urlencode even faster by introducing it's very own modified lookup table.

Commit:
Bump up the version.

ConradIrwin commented:
@kakadiya91 thanks for cleaning this up.

A few comments:
* How much faster does this actually make it?
* It looks like you deleted the tests, can you please re-create them.
* We should delete the buildLookup functions if they're not used at run time to reduce download size

bluthen commented:
Our tests with it showed it around twice as fast. Was nice!

Commit:
Build lookups in a LookupFactory to reduce the download size of the library.

bkakadiya42 commented:
Thank you @ConradIrwin for the review.

> How much faster does this actually make it?

Theoretically, It removes O(n) piece for the URL encoding/decoding and O(c) for the general encoding/decoding. So overall everything still stays O(n).
In practice, for the test case we have in the repo, this makes URL encoding/decoding 10% faster and shaves off single digit ms from the first general encoding/decoding.

> It looks like you deleted the tests, can you please re-create them.

I rewrote the tests, but we still have all of them there in the `test/test.js`. I did remove an unused fixture: `test/sample.js`.

> We should delete the buildLookup functions if they're not used at run time to reduce download size

Thank you, I have moved the building lookups into a separate non-exported file: `lookup_factory.js`. Ref. 4cf4517

Let me know what you think :)
